src/datapool/datapool_shm.c

```diff
@@ -41,3 +41,14 @@ datapool_size(struct datapool *pool)
     return cc_alloc_usable_size(pool);
 }
 
+void
+datapool_set_user_data(const struct datapool *pool, const void *user_data, size_t user_size)
+{
+
```

Comment:
log_error?

Reply:
I am against.
Please see that method **datapool_set_user_data** is called always (in case of shm and in case of pmem implementation) in **_slab_heapinfo_teardown**.
Currently this method do nothing in case of implementation backed by shared memory (shm), real usage of this method is for pmem implementation which supports slab recreation.
**log_error** could give user false information that something is wrong.

Reply:
I find it weird that get_user_data fails with a log_error, but set_user_data fails without...

Feel free to ignore this comment.

Reply:
Yes I am aware of that inconsistency, but currently get_user_data is called only for Pmem implementation and in case if it will be used for shm implementation I want it to give user information that datapool implementation regarding user header must be updated.

Reply:
I agree with Marcin.

Error logs should be user-actionable, not a developer debug information. If you want to ensure that `datapool_get_user_data` is never called in this implementation, change `log_error` to an assert.

Reply:
Done via NOT_REACHED();